File: src/main/java/mmo/core/session/Session.java
package mmo.core.session;

import io.netty.channel.Channel;
import mmo.core.netty.ChannelAdapter;

public interface Session {

    /**
     * Check if the session is alive
     */
    boolean isAlive();

    /**
     * Close the session
     */
    void close();

    /**
     * Handle a received packet
     *
     * @param msg Packet to handle
     */
    void receive(Object msg);

    /**
     * Call when connection is down
     */
    void inactive();

    /**
     * Return IO channel
     */
    ChannelAdapter channel();
}

File: src/main/java/mmo/core/session/Session.kt
package mmo.core.session

import mmo.core.netty.ChannelAdapter

interface Session {
    /**
     * Close the session
     */
    fun close()

    /**
     * Handle a received packet
     *
     * @param msg Packet to handle
     */
    fun receive(msg: Any?)

    /**
     * Call when connection is down
     */
    fun inactive()

    /**
     * Return IO channel
     */
    fun channel(): ChannelAdapter
}